components/ws-manager-mk2/controllers/subscriber_controller.go

```diff
@@ -76,5 +80,21 @@ func (r *SubscriberReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 		}
 	}()
 
-	return c.Watch(source.Kind(mgr.GetCache(), &workspacev1.Workspace{}), &handler.EnqueueRequestForObject{})
+	filterByStatus := predicate.Funcs{
+		CreateFunc: func(ce event.CreateEvent) bool {
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			old := e.ObjectOld.(*workspacev1.Workspace)
+			new := e.ObjectNew.(*workspacev1.Workspace)
+
+			if !cmp.Equal(old.Spec.Ports, new.Spec.Ports) {
+				return true
+			}
```

Comment:
what's the reason for doing the special comparison on ports? Should we do this comparison on the entire spec instead? E.g. there's the workspace timeout values in the spec that can change as well

Reply:
Only the failure to compare Ports introduce errors in the integration tests (changing the visibility requires a notification)